### ISSUE TITLE
`intel` compiler not found in Windows `cmake` project build

### ISSUE THREAD

**Issue by praynaud** [fortran-lang/setup-fortran#45]
# Summary

This is for a Windows `fortran` project that uses `cmake` to build and test and uses `Visual Studio 17 2022` as the generator. I need to install the `intel-classic` compiler, preferably the latest version, which is `2021.10`. The compiler seemingly installs, but then `cmake` does not know the Fortran compiler identification and the build fails.

# Details

My CI script (relevant parts):
```yaml
jobs:
  CI_Windows:
    runs-on: [windows-latest]
    strategy:
      fail-fast: true

    - name: Setup Fortran
      uses: fortran-lang/setup-fortran@v1
      with:
        compiler: intel-classic
        version: '2021.10'

    - name: Check Compilers and Versions
      run: |
        echo $env:FC
        cmake --version
        python --version

    - name: Build and Test
      run: |
        .\build_and_test.bat
```

My `build_and_test.bat` script is as follows:
```cmake
cmake -S . -B build
cmake --build build --config Debug -j %NUMBER_OF_PROCESSORS%
```

The output/error in the GitHub CI is:
```
-- Building for: Visual Studio 17 2022
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 6.2.9200.
-- The Fortran compiler identification is unknown
-- The C compiler identification is MSVC 19.35.32217.1
-- The CXX compiler identification is MSVC 19.35.32217.1
CMake Error at CMakeLists.txt:96 (project):
  No CMAKE_Fortran_COMPILER could be found.
```

I tried other GitHub Actions and the `modflowpy/install-intelfortran-action@v1` action works for my project by takes a very long time. It appears this action is an ancestor of the current `fortran-lang/setup-fortran` action, so I'm hoping there is something in that older action that could be added to the current action to properly set environment variables such that `cmake` can correctly detect the Fortran compiler.

For information, here is the output of my successful CI when using the older Action:
```
-- Building for: Visual Studio 17 2022
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 6.2.9200.
-- The Fortran compiler identification is Intel 2021.7.0.20220726
-- The C compiler identification is MSVC 19.35.32217.1
-- The CXX compiler identification is MSVC 19.35.32217.1
```

Thanks for helping and for creating these great tools!

**Comment by wpbonelli** [comment]
Thanks for raising, ideally this could be tested (planned in #31).

Does anything here help? https://github.com/modflowpy/install-intelfortran-action/issues/10 issues with cmake discovery have been reported before

**Comment by praynaud** [comment]
@wpbonelli Thank you for the links. It does seem like some people have found workarounds by using a different generator. I will investigate and report back if I find a solution.

In the meantime, I fully support the work planned in #31 !

**Comment by praynaud** [comment]
@wpbonelli It's been a minute since I looked into this. Let me give it a try with this new version. I will report back ASAP.

**Comment by praynaud** [comment]
@wpbonelli Unfortunately, the cache restore still does not work. I used the action with `@main` instead of `@v1` (see below). I will let you know tomorrow if a fresh install works, because unfortunately something downstream was wrong with my script upon the first run, so the cache was created but the script failed because of something unrelated. The cache should be 'gone' by tomorrow...
```yaml
jobs:
  CI_Windows:
    runs-on: [windows-latest]
    strategy:
      fail-fast: false

    - name: Setup Fortran
      uses: fortran-lang/setup-fortran@main
      id: setup-fortran
      with:
        compiler: intel-classic
        version: '2021.10'
```

**Comment by praynaud** [comment]
@wpbonelli The fresh install also fails, but I'm now looking into other generators. As mentioned above, the `modflowpy/install-intelfortran-action@v1` action worked on fresh install (not cache restore), so there must be a difference between the two actions that causes the failure for fresh installations using the new action.

**Comment by praynaud** [comment]
@zbeekman I am tagging because you might be able to help...

**Comment by rma-rripken** [comment]
I'm on Windows and using fortran-lang/setup-fortran@main and have a similar problem.  When there isn't a cache the build will work but once the cache is hit cmake fails to find the fortran compiler.   I briefly tried  '-G Ninja' in my local environment and that caused my local build to fail in ways I didn't know how to solve. I was hopeful that #48 was going to fix my build issue but something in my setup still isn't quite right.   I'd appreciate any suggestions or next steps I could try to improve the build configuration.  
https://github.com/rma-rripken/AnnLineGen_RMA

**Comment by djkees** [comment]
I dug into this and can confirm the root cause, plus reproduce it locally outside of CI.

**TL;DR:** `install-intel-windows.bat` always disables Visual Studio integration when installing Intel compilers on Windows. CMake's Visual Studio generator needs that integration to identify a Fortran (or Intel C/C++) compiler, so it never can — regardless of compiler version. Switching to Ninja or NMake Makefiles (as others found) works around it because those generators don't need VS integration at all.

## Root cause

`install-intel-windows.bat` runs the Intel installer with:

    -p=NEED_VS2017_INTEGRATION=0 -p=NEED_VS2019_INTEGRATION=0 -p=NEED_VS2022_INTEGRATION=0

These flags have been present since Windows Intel support was first added (#27) and are unconditional — they disable Visual Studio integration for every Windows Intel install, `intel-classic` and `intel` alike.

CMake's Visual Studio generator identifies a Fortran compiler by generating an old-style `.vfproj` (`ProjectCreator="Intel Fortran"`) and trying to build it. That project format only builds through the Intel Fortran VS-integration plugin — it isn't a native MSBuild project type. With VS integration disabled, there's no handler registered for it, so the identification step fails silently and CMake reports "unknown," which is exactly the symptom in this issue.

## Reproduction

Environment: Windows 10.0.26200, VS Build Tools 2022 17.14 (MSVC 14.44.35207, no full VS IDE — same class of install as GitHub's `windows-latest` runners), CMake 3.31.6 (VS-bundled), Intel oneAPI HPCKit 2023.2.0.

On a clean machine with no prior VS or Intel installs:

1. Installed VS Build Tools 2022 (Desktop development with C++ workload)
2. Installed the Intel oneAPI HPCKit 2023.2.0 `ifort-compiler` + `cpp-compiler` components via this repo's own `install-intel-windows.bat` invocation (`intel-classic` `2021.10` maps to this release)
3. Ran `ifort`/MSVC setup via `setvars.bat --force`, then configured a minimal `project(repro Fortran)` with `cmake -G "Visual Studio 17 2022"`

Result matches this issue exactly:

    -- The Fortran compiler identification is unknown
    CMake Error at CMakeLists.txt:2 (project):
      No CMAKE_Fortran_COMPILER could be found.

CMake's own `CMakeConfigureLog.yaml` confirms the mechanism: the `.vfproj` build fails with `no such file or directory` — there's no Intel Fortran VS-integration handler installed to build it.

## Scope

This affects `ifx` too, not just `intel-classic` — the same three flags are passed for both installers. It's also unchanged in the `v2` rewrite (`develop` branch, #245): `src/installers/ifort/win32.ts` and `src/installers/ifx/win32.ts` both carry the identical flags, so this will still reproduce on `@v2` once it ships.

## Suggested direction

Simply flipping these flags to `1` isn't free — it would add install time/size for every Windows Intel user, including the majority using Ninja/NMake who never hit this. An opt-in input (e.g. `enable-vs-integration: true`) that conditionally drops the `NEED_VS*_INTEGRATION=0` flags seems like the better tradeoff, but that's a design call for the maintainers — happy to put together a PR (against `main` and/or `develop`) if that direction sounds right.

---
Drafted with Claude's assistance.
- Root cause and reproduction steps were independently verified by actually installing the toolchain (VS Build Tools + Intel oneAPI) and running the failing CMake command, not inferred from source alone.
- The claim that `develop`/v2 carries the same flags was confirmed by grepping the actual `origin/develop` branch source, not assumed.

**Comment by wpbonelli** [comment]
Thanks for confirming the cause.

> An opt-in input (e.g. enable-vs-integration: true) that conditionally drops the NEED_VS*_INTEGRATION=0 flags

Seems reasonable, maybe prefix it `intel-enable-vs...` so the same pattern can be used for options for other toolchains, if needed?

**Comment by minhqdao** [comment]
I agree that an extra flag would be necessary for this. I’d like to see this supported in v2 with a proper integration test, but I’d prefer to focus on getting v2 merged first rather than adding this to `develop`.
